src/diffusers/loaders/textual_inversion.py

```diff
@@ -382,14 +376,17 @@ def load_textual_inversion(
             if not isinstance(pretrained_model_name_or_path, list)
             else pretrained_model_name_or_path
         )
-        tokens = len(pretrained_model_name_or_paths) * [token] if (isinstance(token, str) or token is None) else token
+        tokens = [token] if (isinstance(token, str) or token is None) else token
 
         # 3. Check inputs
         self._check_text_inv_inputs(tokenizer, text_encoder, pretrained_model_name_or_paths, tokens)
 
         # 4. Load state dicts of textual embeddings
         state_dicts = load_textual_inversion_state_dicts(pretrained_model_name_or_paths, **kwargs)
 
+        state_dicts = state_dicts * len(tokens)
```

Comment:
Hmm, note that here we're mostly looking for expanding `state_dicts` to be a multiple of tokens, not the other way around.

Reply:
>note that here we're mostly looking for expanding state_dicts to be a multiple of tokens, not the other way around.

that's the current behavior though ("the other way around")  - I'm not sure what a meaningful use case would look like but with current implementation, if we pass a list of `state_dicts` and a single `token`, it will replicate the tokens 
```
[sd1, sd2], token1 
= 
sd1, token1
sd2, token1
```

so now we support a list of tokens as well, I think we would support both, i.e. replicate the tokens for each state_dict and replicate state_dict for each token, unless we do not allow them to multiple pass state_dicts and multiple tokens at the same time? 

```
[sd1, sd2], [token1,token2]
= 
sd1, token1
sd2, token1
sd1, token2
sd2, token2
```

Reply:
@patrickvonplaten 
I'm a little bit bit confused with current implementation (see my comments here https://github.com/huggingface/diffusers/pull/5837#discussion_r1400073856)

I updated the code based on my understanding:
1. expanding `state_dict` to be multiple of `tokens` if the `state_dict` is a tensor that contains same number of embeddings as number of tokens
2. replicate tokens for `state_dicts` only if tokens is `None`

Let me know if this makes sense!